src/characters.jl

```diff
@@ -0,0 +1,171 @@
+    if isone(-M.m[s,t])
+        r = M.r
+        g = first(supp(M.cc[t])) # it doesn't matter which we take
+        M.m[s,t] = (M.cc[r]*M.cc[s])[g] # c_RST = r*s = t
+    # we compute much more above: no we obtain the whole row (? or column?)
+    end
+    return M.m[s,t]
+end
+
+N = 4
+G = PermGroup(N)
+RG = GroupRing(G)
+
+ccG = let ptypes = [p.part for p in AllParts(N)]
+    perms_typed = Dict(g => permtype(g) for g in G)
+    ccG = [Set(g for (g,t) in perms_typed if t==pt) for pt in ptypes]
+    [sum(RG.(cc)) for cc in ccG]
+    # conjugacy classes as all-one elements of GroupRing
+end
+
+M = [CCMatrix(ccG, i) for i in 1:length(ccG)];
+@assert M[1] == one(M[1])
+         
+# <a, b> = sum(a(g^-1)*b(g) for g in RepresentativesConjugacyClasses(G) )
+#
+# ** all calculations can happen over sufficiently large prime p **
+
+
+using DynamicPolynomials
+using MultivariatePolynomials
+
+
+function characteristic_polynomial(M::AbstractArray, T::MultivariatePolynomials.AbstractVariable)
+    n,m = size(M)
+    @assert n==m
+    L = convert.(polynomialtype(T), -M)
+    for i in 1:n
+        L[i,i]+=T
+    end
+    return det(L)
+end
+
+function Base.mod(p::MultivariatePolynomials.AbstractPolynomialLike, m::Int)
+    return dot(mod.(coefficients(p), m), MultivariatePolynomials.monomials(p))
+end
+
+function eval_mod(p::MultivariatePolynomials.AbstractPolynomialLike, m::Int)
+    return [mod(p(i), m) for i = 0:m-1]
+end
+
+function factor_mod(p::MultivariatePolynomials.AbstractPolynomialLike, m::Int)
+    values = eval_mod(p, m)
+    println(values)
+    if all(values.!=0)
+        return p
+    else
+        T = first(variables(p))
+        id = findall(x->x==0, values)
+        factors = [T-mod(i-1,m) for i in id]
+        println(factors)
+        r = p
+        for f in factors
+            r,_ = divrem(r,f)
+            println(r)
+        end
+    end
+    println("here")
+    return [convert.(typeof(p),factors)..., factor_mod(convert(typeof(p),r),m)]
+end
+
+function is_prime(p::Int)
+    if p==1
+        return false
+    else
+        prime = p==2 || p==3
+        if !prime 
+            prime = mod(p,6)==1||mod(p,6)==5
+            if prime 
+                for i = 5:floor(Int, sqrt(p),)
+                    prime = mod(p, i)>0
+                    if !prime
+                        break
+                    end
+                end
+            end
+        end
+        return prime
+    end
+end
+
+function next_prime(z::Int)
+    z += 1
+    while !is_prime(z)
+        z += 1
+    end
+    return z
+end
+
+function find_prime(m::Int) 
+    @assert m < 1000000
+    p = 2
+    while mod(p, m)!=1
+        p = next_prime(p)
+    end
+    return p
+end
+
+function find_int(m::Int, p::Int)
+    z = 2
+    while !(mod(z^m, p)==1) || any([mod(z^m, q)==1 for q in 1:p-1])
+         z+=1
+    end
+    return z
+end
+
+m = Int(lcm(order.(G)))
+q = find_prime(m)
+
+
+@ncpolyvar T
+println(M[2])
+p = mod(characteristic_polynomial(M[2].m, T), q)
```

Comment:
try `p = mod(characteristic_polynomial(M[2], T), q)`

`M[2]` is already an array, so use it as such, never use its internals: `M[2].m` is only initialized to correct values when accessed via `getindex`; `println` asks for values of `M[2]`, so then you get the correct answer;

we may change the name of field from `m` to `lazyclasscoeffs` or something like this.

why T is a `@ncpolyvar` (or what exactly it is?)

with `Nemo` I can do 
```julia
F = Nemo.GF(23)
Fx, x = F["x"]
NM = [matrix(F, m) for m in M]
for m in NM
    @show factor(minpoly(Fx, m))
end
```
```
factor(minpoly(Fx, m)) = 1 * (x+22)
factor(minpoly(Fx, m)) = 1 * (x) * (x+17) * (x+21) * (x+2) * (x+6)
factor(minpoly(Fx, m)) = 1 * (x) * (x+15) * (x+4)
factor(minpoly(Fx, m)) = 1 * (x+20) * (x+1)
factor(minpoly(Fx, m)) = 1 * (x) * (x+17) * (x+21) * (x+2) * (x+6)
```
unfortunately support for factorizations, etc in AA is poor as it's intended to be used as framework for `Nemo`.